test/Microsoft.AspNet.Mvc.FunctionalTests/LocalizationTest.cs

```diff
@@ -71,5 +74,100 @@ public static IEnumerable<object[]> LocalizationData
             // Assert
             Assert.Equal(expected, body.Trim());
         }
+
+        public static IEnumerable<object[]> LocalizationResourceData
+        {
+            get
+            {
+                var expected1 =
+ @"My ASP.NET Application
+
+Hello there!!
+Learn More";
+
+                yield return new[] {"en-GB", expected1 };
+
+                var expected2 =
+ @"Mon application ASP.NET
+
+Bonjour!
+apprendre Encore Plus";
+
+//Coreclr just reads enu resources. Hence the conditional if
+#if DNX451
+                yield return new[] { "fr", expected2 };
+#else
+                yield return new[] {"fr", expected1 };
+#endif
+
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(LocalizationResourceData))]
+        public async Task Localization_Resources(string value, string expected)
```

Comment:
Please add answers to the "does what?" question to the end of all test names e.g. `Localization_Resources_ReturnExpectedValues` here.